### PR TITLE
Fix BQ query function with projected columns

### LINKED ISSUE #19570: Projection with `query` pass-through function may return incorrect result in BigQuery connector
The following query should return 2. 
```sql
select y from ( select x , y ,z from table(bigquery.system.query(query => 'select 1 x, 2 y, 3 z')));
 y
---
 1
(1 row)
```

### PR DESCRIPTION
**Pull request by wendigo**
## Description

Fixes #19570

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by wendigo** [comment]
BQ module timed out but the new test has passed:
```
2023-10-30T13:40:53.1506173Z 2023-10-30T07:40:53.148-0600	INFO	pool-3-thread-2	io.trino.testng.services.ProgressLoggingListener	[TEST SUCCESS] io.trino.plugin.bigquery.TestBigQueryAvroConnectorTest.testNativeQuerySimpleWithProjectedColumns; (took: 1.8 seconds)

```

**Comment by wendigo** [comment]
I'm going to restart BQ either way

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryQueryPageSource.java [hunk 1]
```diff
@@ -102,11 +103,14 @@ public BigQueryQueryPageSource(
 
     private static String buildSql(BigQueryTableHandle table, String projectId, List<String> columnNames, Optional<String> filter)
     {
+        String columns = columnNames.stream().map(column -> format("`%s`", column)).collect(joining(","));
```
Comment (ebyhr) [inline plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryQueryPageSource.java:106]: nit: We can use `BigQueryUtil::quote` method. Also, no need to build `columns` when `columnNames` is empty. 
```java
String projection = columnNames.isEmpty() ? "*" : columnNames.stream().map(BigQueryUtil::quote).collect(joining(","));
```

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryQueryPageSource.java [hunk 2]
```diff
@@ -102,11 +103,14 @@ public BigQueryQueryPageSource(
...
+        String projection = columnNames.isEmpty() ? "*" : columns;
```
Comment (ebyhr) [inline plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryQueryPageSource.java:107]: Is the first condition `columnNames.isEmpty()` covered by any tests?

File: plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryConnectorTest.java [hunk 3]
```diff
@@ -803,6 +803,18 @@ public void testNativeQuerySimple()
...
 
+    @Test
+    public void testNativeQuerySimpleWithProjectedColumns()
+    {
+        assertQuery(
+                "SELECT z, y, x FROM (SELECT y, z, x FROM TABLE(bigquery.system.query(query => 'select 1 x, 2 y, 3 z')))",
+                "VALUES (3, 2, 1)");
+
+        assertQuery(
+                "SELECT z FROM (SELECT x, y, z FROM TABLE(bigquery.system.query(query => 'select 1 x, 2 y, 3 z')))",
```
Comment (ebyhr) [inline plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryConnectorTest.java:814]: nit: Uppercase `select` in query parameter.

**Review by ebyhr — APPROVED** [review]

**Review by ebyhr — CHANGES_REQUESTED** [review]
This change causes query failure when the `query` argument contains implicit column names. e.g.
```sql
SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT 1'));
```

**Comment by wendigo** [comment]
@ebyhr yeah, I've noticed

**Pull request closed without merging**